### PR TITLE
Cleanup completed gitjobs

### PR DESCRIPTION
**Pull request by manno**
Refers to https://github.com/rancher/fleet/issues/2870

This is a workaround until the gitops controller removes jobs as needed in a future release.

## Testing

After creating several commits in a git repo, observe there are many completed jobs. One per commit.
Change the new CronJob schedule to minutes: `* */1 * * *`, observe that all but the latest job are removed for each gitrepo .

### CODE REVIEW

File: internal/cmd/cli/cleanup/cleanup.go [hunk 1]
```diff
@@ -123,13 +124,97 @@ func ClusterRegistrations(ctx context.Context, client Getter, opts Options) erro
...
+			})
+
+			// if there is a running job delete all the jobs in the
+			// list, otherwise all but the newest
+			start := 1
+			if _, ok := running[ns][gitrepo]; ok {
+				start = 0
+			}
+
+			logger.V(1).Info("Deleting jobs for gitrepo", "n", len(jobs)-start, "namespace", ns, "gitrepo", gitrepo)
```
Comment (weyfonk) [inline internal/cmd/cli/cleanup/cleanup.go:207]: Would we want to make this more precise, or would we simply rely on logs from requests made by the caller function?
```suggestion
			logger.V(1).Info("Deleting jobs for gitrepo", "n", len(jobs)-start, "total", len(jobs), "namespace", ns, "gitrepo", gitrepo)
```

**Review by weyfonk — COMMENTED** [review]
This almost LGTM, pending a doubt on client instantiation. Not a functionality nor performance concern.

File: internal/cmd/cli/cleanup/cleanup.go [hunk 2]
```diff
@@ -123,13 +124,97 @@ func ClusterRegistrations(ctx context.Context, client Getter, opts Options) erro
...
+	gitjobs := map[string]map[string][]batchv1.Job{}
+	// gitrepos with running jobs
+	running := map[string]map[string]struct{}{}
+
+	for _, job := range jobs {
+		if job.OwnerReferences == nil {
+			continue
+		}
+		for _, or := range job.OwnerReferences {
+			if or.Kind == "GitRepo" && or.APIVersion == "fleet.cattle.io/v1alpha1" {
```
[the remark this answers is not in the document]
Comment (manno) [reply internal/cmd/cli/cleanup/cleanup.go:177]: Ah, nice.

File: internal/cmd/cli/cleanup/cleanup.go (hunk 1 above)
Comment (manno) [reply internal/cmd/cli/cleanup/cleanup.go:207]: I'm not sure what you want to log?

Comment (weyfonk) [reply internal/cmd/cli/cleanup/cleanup.go:208]: The idea was to log the total amount of jobs being processed, to be able to follow the deletion job's progress through logs, but that's not critical I guess.

**Pull request merged**